test/e2e/framework/framework.go

```diff
@@ -292,6 +292,7 @@ func (f *Framework) WaitForGameServerState(t *testing.T, gs *agonesv1.GameServer
 // Each Allocated GameServer gets deleted allocDuration after it was Allocated.
 // GameServers will continue to be Allocated until a message is passed to the done channel.
 func (f *Framework) CycleAllocations(ctx context.Context, t *testing.T, flt *agonesv1.Fleet, period time.Duration, allocDuration time.Duration) {
+	// nolint:staticcheck
 	err := wait.PollImmediateUntil(period, func() (bool, error) {
```

Comment:
This can also be replaced with [PollUntilContextCancel](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediateUntil)